libnmstate/nm/device.py

```diff
@@ -149,3 +154,27 @@ def get_device_common_info(dev):
 def is_externally_managed(nmdev):
     nm_ac = nmdev.get_active_connection()
     return nm_ac and NM.ActivationStateFlags.EXTERNAL & nm_ac.get_state_flags()
+
+
+def mark_device_as_managed(context, nm_dev):
+    action = f"Set device as managed: {nm_dev.get_iface()}"
+    context.register_async(action, fast=True)
+    user_data = context, action
+    context.client.dbus_set_property(
+        NM.Object.get_path(nm_dev),
+        NM_DBUS_INTERFACE_DEVICE,
+        "Managed",
+        GLib.Variant.new_boolean(True),
+        NM_USE_DEFAULT_TIMEOUT_VALUE,
+        context.cancellable,
+        _set_managed_callback,
+        user_data,
```

Comment:
Is this the only way possible? It seems quite confusing for me. In my opinion, NM should expose something like `set_managed_async()`. If that is the only way possible now, let's approve it but I would consider replace it in the future :-)

Reply:
Yes. That's the only way. :(